pkg/module/go_vcs_lister.go

```diff
@@ -30,43 +21,25 @@ type listResp struct {
 }
 
 type vcsLister struct {
-	goBinPath string
-	goProxy   string
-	fs        afero.Fs
+	fs afero.Fs
+
+	goTool
 }
 
 func (l *vcsLister) List(ctx context.Context, mod string) (*storage.RevInfo, []string, error) {
 	const op errors.Op = "vcsLister.List"
 	ctx, span := observ.StartSpan(ctx, op.String())
 	defer span.End()
 
-	tmpDir, err := afero.TempDir(l.fs, "", "go-list")
-	if err != nil {
-		return nil, nil, errors.E(op, err)
-	}
-	defer l.fs.RemoveAll(tmpDir)
-
-	cmd := exec.Command(
-		l.goBinPath,
-		"list", "-m", "-versions", "-json",
-		config.FmtModVer(mod, "latest"),
-	)
-	cmd.Dir = tmpDir
-	stdout := &bytes.Buffer{}
-	stderr := &bytes.Buffer{}
-	cmd.Stdout = stdout
-	cmd.Stderr = stderr
-
-	gopath, err := afero.TempDir(l.fs, "", "athens")
+	runtime, err := prepareRuntime(l.fs, l.goTool, "")
 	if err != nil {
 		return nil, nil, errors.E(op, err)
 	}
-	defer module.ClearFiles(l.fs, gopath)
-	cmd.Env = module.PrepareEnv(gopath, l.goProxy)
+	defer runtime.clean()
 
-	err = cmd.Run()
+	err = runtime.run("list", "-m", "-versions", "-json", config.FmtModVer(mod, "latest"))
```

Comment:
this could be called `gocmd.run(...)` but i still prefer us to flat out the logic instead of abstract it.